src/software/ai/passing/base_pass.h

```diff
@@ -0,0 +1,109 @@
+#pragma once
+
+#include <array>
+#include <cstdlib>
+#include <iostream>
+
+#include "software/geom/point.h"
+#include "software/time/duration.h"
+
+
+// The number of parameters (representing a pass) that we optimize
+// (receive_location_x, receive_location_y)
+static const int NUM_PARAMS_TO_OPTIMIZE = 2;
+
+class BasePass 
+{
+    public:
+    BasePass() = delete;
+
+    /**
+     * Gets the value of the passer point
+     *
+     * @return The value of the passer point
+     */
+    Point passerPoint() const;
+
+    /**
+     * Gets the value of the receiver point
+     *
+     * @return The value of the receiver point
+     */
+    Point receiverPoint() const;
+
+    /**
+     * Given the ball position, returns the angle the receiver should be
+     * facing to receive the pass.
+     *
+     * @return The angle the receiver should be facing
+     */
+    Angle receiverOrientation() const;
+
+    /**
+     * Given the ball position, returns the angle the passer should be
+     * facing to pass.
+     *
+     * @return The angle the passer should be facing
+     */
+    Angle passerOrientation() const;
+
+    /**
+     * Gets the length of the pass in metres
+     *
+     * @return The length of the pass in metres
+     */
+    double length() const;
+
+    /**
+     * Estimate how long the pass will take, from kicking to receiving
+     *
+     * This estimate does not account for friction on the ball
+     *
+     * @return An estimate of how long the pass will take, from kicking to receiving
+     */
+    virtual Duration estimatePassDuration() const
+    {
+        return Duration::fromSeconds(0);
+    }
```

Comment:
this seems a bit of a useless implementation, why not do `virtual Duration estimatePassDuration() const = 0`?